uptime.go

```diff
@@ -0,0 +1,324 @@
+package godo
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"path"
+)
+
+const uptimeChecksBasePath = "/v2/uptime/checks"
+
+// UptimeChecksService is an interface for creating and managing Uptime checks with the DigitalOcean API.
+// See: https://docs.digitalocean.com/reference/api/api-reference/#tag/Projects
+type UptimeChecksService interface {
+	List(context.Context, *ListOptions) ([]UptimeCheck, *Response, error)
+	Get(context.Context, string) (*UptimeCheck, *Response, error)
+	GetState(context.Context, string) (*UptimeCheck, *Response, error)
```

Comment:
The state response body is not the same as an `UptimeCheck`. We'll need a new type for this.

Here's an example response from one of my checks:

```json
{
  "state": {
    "alert_count": 1,
    "days_to_ssl_expiry": 279,
    "regions": {
      "eu_west": {
        "status": "UP",
        "status_changed_at": "2022-07-23T11:19:35Z",
        "thirty_day_uptime_percentage": 99.99
      },
      "se_asia": {
        "status": "UP",
        "status_changed_at": "2022-06-21T07:01:45Z",
        "thirty_day_uptime_percentage": 99.99
      },
      "us_east": {
        "status": "UP",
        "status_changed_at": "2022-06-21T07:05:18Z",
        "thirty_day_uptime_percentage": 99.99
      },
      "us_west": {
        "status": "UP",
        "status_changed_at": "2022-11-01T06:27:46Z",
        "thirty_day_uptime_percentage": 99.99
      }
    },
    "previous_outage": {
      "region": "us_west",
      "started_at": "2022-11-01T06:26:46Z",
      "ended_at": "2022-11-01T06:27:46Z",
      "duration_seconds": 60
    }
  }
}
```

https://docs.digitalocean.com/reference/api/api-reference/#operation/uptime_check_state_get